### ISSUE TITLE
Unexpected end of JSON input

### ISSUE THREAD

**Issue by federico-fiorini** [segmentio/consent-manager#91]
### Description
After upgrading to the latest version of the consent manager (4.5.0) we started experiencing a spike in errors that seem to be happening within the consent manager code.

<img width="1013" alt="Screen Shot 2020-04-08 at 19 12 07" src="https://user-images.githubusercontent.com/5628190/78846882-d90fb600-79d2-11ea-90e7-275ce1a8e5b5.png">

Something I noticed is that when opening the consent manager, the preferences are not set - even though I had set them previously and the cookie is set correctly. So this makes me thing there might be an issue related to how the consent manager code reads the cookie. However, after closing it and re-opening again, the options are now set correctly
![consent-manager-debugging](https://user-images.githubusercontent.com/5628190/78847470-81724a00-79d4-11ea-88f6-2224860a64da.gif)

#### Notes
We upgraded from an old version: 1.1.1

**Comment by danielstjules** [comment]
Hi @federico-fiorini! To confirm, you're using the react components rather than the standalone script? Does this same issue exist with the standalone?

Is this code active on an URL for testing or to help reproduce?

**Comment by federico-fiorini** [comment]
Hi @danielstjules! I think we're using the standalone. Our script looks like this:

```
<script>
window.consentManagerConfig = function(exports) {
  var React = exports.React
  var bannerContent = ' ' 
  var preferencesDialogTitle = 'Cookie Preferences'
  var preferencesDialogContent = 'We use data collected by cookies and JavaScript libraries to improve the functionality of the Webflow Dashboard, deliver personalized advertisements, and increase the overall performance of our service.'
  var cancelDialogTitle = 'Are you sure you want to cancel?'
  var cancelDialogContent = React.createElement(
    'span',
    null,
    'Your preferences have not been saved. By continuing to use our website, you՚re agreeing to our',
    ' ',
    React.createElement(
      'a',
      {href: '/legal/cookie-policy', target: '_blank'},
      'Cookie Policy'
    ),
    '.'
  )
  var defaultDestinationBehavior = 'imply'

  return {
    container: '#consent-container',
    writeKey: '#{API_TOKEN}',
    shouldRequireConsent: function () { return false; },
    bannerContent: bannerContent,
    preferencesDialogTitle: preferencesDialogTitle,
    preferencesDialogContent: preferencesDialogContent,
    cancelDialogTitle: cancelDialogTitle,
    cancelDialogContent: cancelDialogContent,
    defaultDestinationBehavior: defaultDestinationBehavior
  }
}
</script>

<script src='https://unpkg.com/@segment/consent-manager@4.5.0/standalone/consent-manager.js' integrity='sha384-g/zCt4IB4iphg6Jsy8xsUoBftTDwbC9AsJLGlQEtCIs4b4s4aLBk80RmNUBqXOvp' crossorigin='anonymous' defer></script>

<div id="consent-container"></div>
```

It's active on our dashboard pages (which is where we see the errors being reported) e.g. https://webflow.com/dashboard however I wasn't able to reproduce on my end

**Comment by federico-fiorini** [comment]
@danielstjules by looking at the stacktrace of the error it seems like it's happening [here](https://github.com/segmentio/consent-manager/blob/master/src/consent-manager-builder/fetch-destinations.ts#L14)? Is it possible that the integration endpoint returns a badly formatted json response? Could this have anything to do with ad blockers?

**Comment by danielstjules** [comment]
Thanks! Tested on the site, and signed up to view the dashboard as an auth'ed user and was unable to reproduce. 

Can you confirm that you're able to load the following URL? https://cdn.segment.com/v1/projects/j8cpth9hiw/integrations If you're able to reproduce this issue your own machine, could you open a support ticket and send us a HAR file containing the network requests that led to the error.

Separately, if you're not able to reproduce yourself, this might be a case of ad blockers or other browser security settings blocking the request. You can catch this and treat as you'd like (suppress, log, emit another stat) with the `onError` handler:

https://github.com/segmentio/consent-manager/blob/41b480235cb01606dfe67f902992467e8305de38/src/consent-manager/index.tsx#L48

**Comment by danielstjules** [comment]
Going to https://webflow.com/ it seems like you might be loading ajs twice?

<img width="850" alt="Screen Shot 2020-04-15 at 9 33 17 PM" src="https://user-images.githubusercontent.com/817212/79414927-d11abd80-7f60-11ea-9cc8-65e1fe58b09e.png">

**Comment by danielstjules** [comment]
@federico-fiorini Regarding:
> Something I noticed is that when opening the consent manager, the preferences are not set - even though I had set them previously and the cookie is set correctly

Team has been able to replicate and are narrowing down on the exact cause of the bug!

**Comment by federico-fiorini** [comment]
@danielstjules thanks!

> if you're not able to reproduce yourself, this might be a case of ad blockers or other browser security settings blocking the request

That was my initial thought too but I wasn't able to reproduce even by enabling ad blockers. The /integrations request is blocked but it doesn't error with `Unexpected end of JSON input`. In fact the consent manager code seems to handle failed requests. It seems to me that this error would happen if the request is successful but returns a bad response. I wonder if certain ad blockers instead of blocking requests can override responses.

> The error could be bubbling up because we're now awaiting on init

That makes sense

> You can catch this and treat as you'd like with the onError handler

Okay good to know!

> Going to https://webflow.com/ it seems like it might be loading ajs twice, but with different sources

Oh that's weird! I'm gonna look into it. Though I checked and the two keys belong to the same source

> Team has been able to replicate and are narrowing down on the exact cause of the bug!

Great! And I assume this has nothing to do with the other error bubbling up

**Comment by notfelineit** [comment]
Hey @federico-fiorini :-) 

Regarding:
> Something I noticed is that when opening the consent manager, the preferences are not set - even though I had set them previously and the cookie is set correctly

We just pushed a fix here: https://github.com/segmentio/consent-manager/pull/95 and have released a new version, 4.5.1. Thank you for your patience!

**Comment by federico-fiorini** [comment]
Hi @notfelineit 👋Thanks for letting me know! I'll update to the new version
